Runtime/Model/DataProvider/UnityMachineIdentifierProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Backtrace.Unity.Extensions;
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+using UnityEngine;
+
+namespace Backtrace.Unity.Model.DataProvider
+{
+    internal class UnityMachineIdentifierProvider : IMachineIdentifierDataProvider
+    {
+        private readonly string _deviceUniqueIdentifier;
+        internal UnityMachineIdentifierProvider() : this(SystemInfo.deviceUniqueIdentifier) { }
+
+        internal UnityMachineIdentifierProvider(string machineIdentifier)
+        {
+            _deviceUniqueIdentifier = machineIdentifier;
+        }
+        public string Get()
+        {
+            if (!IsValidIdentifier())
+            {
+                return null;
+            }
+
+            if (Guid.TryParse(_deviceUniqueIdentifier, out Guid unityUuidGuid))
+            {
+                return unityUuidGuid.ToString();
+            }
+            return GuidHelper.FromString(_deviceUniqueIdentifier).ToString();
```

Comment:
How often the ID will be generated? Maybe it's a good idea to generate this in the constructor, or cache the value somewhere?

Reply:
Oh, I see that in `MachineIdStorage` it is retrieved only once? If so, feel free to discard this comment.

Reply:
to confirm - yes. we're doing it only once.